modules/run-vault/run-vault

```diff
@@ -134,6 +151,75 @@ function vault_version_at_least {
   fi
 }
 
+function generate_vault_agent_config {
+  local -r config_dir="$1"
+  local -r data_dir="$2"
+  local -r vault_address="$3"
+  local -r vault_port="$4"
+  local -r ca_cert_file="$5"
+  local -r client_cert_file="$6"
+  local -r client_key_file="$7"
+  local -r auth_mount_path="$8"
+  local -r auth_type="$9"
+  local -r auth_role="${10}"
+
+  local -r config_path="$config_dir/$VAULT_CONFIG_FILE"
+
+  log_info "Creating default Vault Agent config file in $config_path"
+
+  local -r pid_config="pid_file   = \"$data_dir/$VAULT_PID_FILE\""
+
+  local ca_cert_config=""
+  if [[ ! -z $ca_cert_file ]]; then
+   ca_cert_config=$(cat <<EOF
+  ca_cert = "$ca_cert_file"\n
+EOF
+)
+  fi
+
+  local client_cert_config=""
+  if [[ ! -z $client_cert_file ]] && [[ ! -z $client_key_file ]]; then
+   client_cert_config=$(cat <<EOF
+  client_cert = "$client_cert_file"
+  client_key = "$client_key_file"\n
+EOF
+)
+  fi
+
+  local -r vault_config=$(cat <<EOF
+vault {
+  address = "https://$vault_address:$vault_port"
+$ca_cert_config
+$client_cert_config
+}\n
+EOF
+)
+
+  local -r auto_auth_config=$(cat <<EOF
+auto_auth {
+  method "aws" {
+    mount_path = "$auth_mount_path"
+    config = {
+      type = "$auth_type"
+      role = "$auth_role"
+    }
+  }
+
+  sink "file" {
+    config = {
+      path = "$data_dir/$VAULT_TOKEN_FILE"
+    }
+  }
```

Comment:
Oh, I see, you've added that file specifically as a sink. OK, up above in the script that reads this sink, please add comments explaining that!

Reply:
Added a comment in the script, as well as a description in the README.